discovery/README.md

```diff
@@ -121,3 +121,42 @@ or incorrect metadata.
 The information obtained from service discovery is not considered sensitive
 security wise. Do not return secrets in metadata, anyone with access to
 the Prometheus server will be able to see them.
+
+
+## Writing an SD mechanism
+
+### The SD Interface
```

Comment:
Interface -> interface

...since we're not using title case anywhere else in the doc.